types/src/transaction/mod.rs

```diff
@@ -971,10 +971,14 @@ impl TransactionStatus {
         }
     }
 
-    pub fn from_vm_status(vm_status: VMStatus, charge_invariant_violation: bool) -> Self {
+    pub fn from_vm_status(
```

Comment:
as discussed offline: would make a lot of sense if this consumes vm_status and outputs both the txn_status and the aux_data, and optionally, error details are removed from the txn_status.